extension/registry.go

```diff
@@ -43,4 +43,6 @@ type Registry struct {
 	// NewKeyProto creates a new private key based on a key specification.
 	// It returns a proto that can be passed to a keys.ProtoHandler to get a crypto.Signer.
 	NewKeyProto keys.ProtoGenerator
+	// SetProcessStatus sets the current process status for diagnostic purposes.
+	SetProcessStatus func(string)
```

Comment:
This seems more like a sequencer-specific feature and less like an extension point. WDYT about moving it to `LogOperationManager`?

Reply:
I was thinking of this as a generic extension point -- there are other potential uses across the different jobs (`SetProcessStatus("terminating due to SIGQUIT")`, `SetProcessStatus("starting up")`, ...)